libs/utils/test.py

```diff
@@ -100,4 +102,24 @@ def _experimentsFinalize(cls):
         Code executed after running the experiments
         """
 
+@wrapt.decorator
+def experiment_test(wrapped_test, instance, args, kwargs):
+    """
+    Convert a LisaTest test method to be automatically called for each experiment
+
+    The method will be passed the experiment object and a list of the names of
+    tasks that were run as the experiment's workload.
+    """
+    for experiment in instance.executor.experiments:
+        tasks = experiment.wload.tasks.keys()
+        try:
+            wrapped_test(experiment, tasks, *args, **kwargs)
+        except AssertionError as e:
+            trace_relpath = os.path.join(experiment.out_dir, "trace.dat")
+            e.args += ("Check trace file: " + os.path.abspath(trace_relpath),)
```

Comment:
Maybe a possible alternative solution could be to split the 'Check trace...' message as a WARNING log message right before raising the exception.

I would expect in that case we should get the trace path right before the Traceback.

You can also split some useful "\n\n\n" right at the end of the warning message to keep some space before the traceback...